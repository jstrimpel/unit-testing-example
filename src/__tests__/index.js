import React from 'react';
import renderer from 'react-test-renderer';
import Index from '../index.js';
import sinon from 'sinon';
import { shallow } from 'enzyme';

describe('Index Component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Index title={'I am an example.'} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should respond to a user click', () => {
    const spy = sinon.spy();
    const component = shallow(<Index title={'I am an example.'} onClick={spy}/>);

    component.find('p').simulate('click', {});    

    expect(spy.callCount).to.equal(1);
  });
});