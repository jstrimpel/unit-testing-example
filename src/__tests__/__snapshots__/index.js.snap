// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Component should render 1`] = `
<p
  onClick={undefined}
>
  I am an example.
</p>
`;
